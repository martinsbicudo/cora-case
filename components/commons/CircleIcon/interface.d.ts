export interface CircleIconProps {
  type?: 'credit' | 'debit' | 'refunded'
}
