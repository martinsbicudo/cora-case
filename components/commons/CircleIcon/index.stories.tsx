import { StoryObj, Meta } from '@storybook/react'

import CircleIcon from '.'

export default {
  title: 'Components / CircleIcon',
  component: CircleIcon,
  argTypes: {
    type: {
      control: 'select',
      defaultValue: 'credit',
      options: ['credit', 'debit', 'refunded'],
    },
  },
} as Meta

export const Default: StoryObj = {}
