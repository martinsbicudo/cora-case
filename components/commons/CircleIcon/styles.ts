import styled from 'styled-components'

export const CircleIcon = styled.i`
  display: flex;
  align-items: center;
  justify-content: center;
  width: 32px;
  height: 32px;
  background-color: ${({ theme }) => theme.colors['soft-secondary']};
  border-radius: 16px;
`
