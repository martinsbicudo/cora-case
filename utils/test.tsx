import type { ReactElement, ReactNode } from 'react'

import { ThemeProvider } from 'styled-components'

import theme from '@/styles/theme'
import { render, RenderOptions } from '@testing-library/react'

const Providers = ({ children }: { children: ReactNode }) => {
  return <ThemeProvider theme={theme}>{children}</ThemeProvider>
}

const customRender = (
  ui: ReactElement,
  options?: Omit<RenderOptions, 'wrapper'>
) => render(ui, { wrapper: Providers, ...options })

export * from '@testing-library/react'
export { customRender as render }
