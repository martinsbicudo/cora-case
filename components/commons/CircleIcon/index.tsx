import { ReactComponent as ArrowDownIn } from '@/public/icons/arrow-down-in.svg'
import { ReactComponent as ArrowReturn } from '@/public/icons/arrow-return.svg'
import { ReactComponent as TimeClock } from '@/public/icons/time-clock.svg'

import { CircleIconProps } from './interface'
import * as Styled from './styles'

const CircleIcon = ({ type = 'credit' }: CircleIconProps) => {
  const renderIcon = () => {
    const Icon = {
      credit: ArrowDownIn,
      debit: ArrowReturn,
      refunded: TimeClock,
    }[type]

    return <Icon />
  }

  return (
    <Styled.CircleIcon data-testid={type}>{renderIcon()}</Styled.CircleIcon>
  )
}

export default CircleIcon
