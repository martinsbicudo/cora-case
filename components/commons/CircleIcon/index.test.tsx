import { render, screen } from '@/test/helpers'

import CircleIcon from '.'
import { CircleIconProps } from './interface'

describe('Components / CircleIcon', () => {
  const renderComponent = (type?: CircleIconProps['type']) =>
    render(<CircleIcon type={type} />)

  it('should render component', () => {
    const { container } = renderComponent()
    expect(container).toMatchSnapshot()
  })

  it('should find credit type icon', () => {
    const type = 'credit'
    renderComponent(type)

    const icon = screen.getByTestId(type)
    expect(icon).toBeInTheDocument()
  })

  it('should find debit type icon', () => {
    const type = 'debit'
    renderComponent(type)

    const icon = screen.getByTestId(type)
    expect(icon).toBeInTheDocument()
  })

  it('should find refunded type icon', () => {
    const type = 'refunded'
    renderComponent(type)

    const icon = screen.getByTestId(type)
    expect(icon).toBeInTheDocument()
  })
})
