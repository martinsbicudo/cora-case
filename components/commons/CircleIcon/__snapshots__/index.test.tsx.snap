// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Components / CircleIcon should render component 1`] = `
<div>
  <i
    class="sc-bcXHqe gZLDnf"
    data-testid="credit"
  >
    <i>
      icon-mock
    </i>
  </i>
</div>
`;
